go/client/versionfix.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package client
+
+import (
+	"github.com/blang/semver"
+	"github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol"
+	"golang.org/x/net/context"
+	"os"
+	"time"
+)
+
+func FixVersionClash(g *libkb.GlobalContext, cl libkb.CommandLine) (err error) {
+	var cli keybase1.ConfigClient
+	var ctlCli keybase1.CtlClient
+	var serviceConfig keybase1.Config
+
+	g.Log.Debug("+ FixVersionClash")
+	defer func() {
+		g.Log.Debug("- FixVersionClash -> %v", err)
+	}()
+	cli, err = GetConfigClient(g)
+	if err != nil {
+		return err
+	}
+	serviceConfig, err = cli.GetConfig(context.TODO(), 0)
+	if err != nil {
+		return err
+	}
+	g.Log.Debug("| Contacted service; got version: %s", serviceConfig.Version)
+
+	// We'll check and restart the service if there is a new version.
+	var semverClient, semverService semver.Version
+
+	cliVersion := libkb.VersionString()
+	if g.Env.GetRunMode() == libkb.DevelRunMode {
+		tmp := os.Getenv("KEYBASE_SET_VERSION")
+		if len(tmp) > 0 {
+			cliVersion = tmp
+		}
+	}
+
+	semverClient, err = semver.Make(cliVersion)
+	if err != nil {
+		return err
+	}
+	semverService, err = semver.Make(serviceConfig.Version)
+	if err != nil {
+		return err
+	}
+
+	g.Log.Debug("| version check %s v %s", semverClient, semverService)
+	if !semverClient.GT(semverService) {
```

Comment:
But then a restart wouldn't fix anything.  The same condition would apply the next time.  Better to just leave it alone, no?